frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -47,55 +54,53 @@ const DataListRow: React.FC<DataListRowProps> = ({ column, onChange, disableUnch
 export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
   cancel,
   close,
-  columnManagementID,
+  columnManagementId,
   columnManagementType,
   selectedColumns,
 }) => {
   const dispatch = useDispatch();
-  const initialDefaultColumns = !_.isEmpty(selectedColumns)
-    ? selectedColumns.filter((column) => column.title.length > 0 && !column.additional)
-    : {};
-  const initialAdditionalColumns = !_.isEmpty(selectedColumns)
-    ? selectedColumns.filter((column) => column.additional)
-    : {};
-  const [defaultColumns, setDefaultColumns] = React.useState<any[]>(initialDefaultColumns);
-
-  const [additionalColumns, setAdditionalColumns] = React.useState<ColumnHeaderRow[]>(
-    initialAdditionalColumns,
+  const initialSelected = new Set<string>(
+    selectedColumns.filter((column) => column.visible && column.id).map((column) => column.id),
   );
+  const defaultColumns = !_.isEmpty(selectedColumns)
+    ? selectedColumns.filter((column) => column.id && !column.additional)
+    : [];
+  const additionalColumns = !_.isEmpty(selectedColumns)
+    ? selectedColumns.filter((column) => column.additional)
+    : [];
 
-  const updateColumns = (checked: boolean, id: string, columnArray: any): any => {
-    return columnArray.map((column) => {
-      if (!id || column.ID === id) {
-        return { ...column, visible: checked };
-      }
-      return column;
-    });
-  };
+  const [checkedColumns, setCheckedColumns] = React.useState<Set<string>>(new Set(initialSelected));
```

Comment:
nit: TypeScript can infer the `useState` type as long as it's passed a value.

```suggestion
  const [checkedColumns, setCheckedColumns] = React.useState(new Set(initialSelected));
```

Reply:
removed type here